### ISSUE TITLE
`odo dev --no-watch`

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#5631]
## User Story
- As an odo user
- I want to be able to run `odo dev` command without triggering push with when files change
- So that the sync and application restart is not constantly happening when I'm making a lot of changes in my ide/editor.

note: There is a followup story that will allow users to trigger push "on-demand" https://github.com/redhat-developer/odo/issues/5634

## Acceptance Criteria
- [ ] `odo dev` command should have `--no-watch` flag. When this flag is specified the command should not watch files for changes.
- [ ] the push happens only once when the command is executed

/kind user-story
